### PR TITLE
docs(adr-116): modular crypto

### PR DESCRIPTION
**Pull request by melekes**
Refs #2424

---

#### PR checklist

- [ ] ~~Tests written/updated~~
- [ ] ~~Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)~~
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

**Comment by zmanian** [comment]
Many of the advantages of alternate cryptography require more invasive changes to the code base and even the header structure.

For instance to get advantages from BLS-12-381, you also need to be able use aggregated public keys in the commit certificates.

There are wins like a more EVM verification friendly comet that uses keccack and ethsign for signatures that can maybe be done without a hard fork.

File: docs/references/architecture/adr-116-modular-crypto.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    GenPrivKey() PrivKey
+}
+
+type SignatureScheme uint
+
+const (
+	Ed25519 SignatureScheme = 1 + iota
+	Secp256k1
+	Sr25519
+	Bls12381 // import github.com/cometbft/bls12381
```
Comment (sergio-mena) [inline docs/references/architecture/adr-116-modular-crypto.md:42]: I'd rather use the _KeyType_ (a string) that every crypto package must define, and which is supposed to be unique. Examples [here](https://github.com/cometbft/cometbft/blob/10bfd89d/crypto/secp256k1/secp256k1.go#L24), [here](https://github.com/cometbft/cometbft/blob/10bfd89d/crypto/ed25519/ed25519.go#L62), or [here](https://github.com/cometbft/cometbft/blob/d27732ad/crypto/bls12381/const.go#L11).

**Review by sergio-mena — COMMENTED** [review]
Although still very preliminary, I support this approach

**Comment by melekes** [comment]
I will reopen once this work is restarted.

**Pull request closed without merging**